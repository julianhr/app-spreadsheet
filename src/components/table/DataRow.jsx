import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'

import RowHeader from '../cell/RowHeader'
import ResultCell from '../ResultCell/ResultCell'


const Row = styled.div`
  display: flex;
  width: fit-content;
`

function DataRow({ rowNumber, colLabels }) {
  const dataCells = () => (
    new Array(colLabels.length).fill(0).map((_, i) => {
      const location = `${colLabels[i]}-${rowNumber}`
      return (
        <ResultCell
          key={i}
          location={location}
        />
      )
    })
  )

  return (
    <Row
      data-row={rowNumber}
    >
      <RowHeader
        label={rowNumber}
      />
      {dataCells()}
    </Row>
  )
}

DataRow.propTypes = {
  rowNumber: PropTypes.number,
  colLabels: PropTypes.arrayOf(PropTypes.string),
}

export default DataRow

