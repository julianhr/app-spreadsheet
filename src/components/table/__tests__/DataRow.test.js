import React from 'react'
import { create } from 'react-test-renderer'
import { render, fireEvent, cleanup } from 'react-testing-library'

import MockApp from '~/__tests__/__mocks__/MockApp'
import DataRow from '../DataRow'


const renderApp = (props) => {
  const wrapper = render(
    <MockApp>
      <table>
        <DataRow {...props} />
      </table>
    </MockApp>
  )

  return wrapper
}

describe('DataRow', () => {
  afterEach(cleanup)

  it('matches snapshot', () => {
    const props = {
      rowNumber: 2,
      colLabels: ['A', 'B'],
      activeCell: 'B-2'
    }
    renderApp(props)
    const el = document.querySelector('tr')
    expect(el).toMatchSnapshot()
  })
})

