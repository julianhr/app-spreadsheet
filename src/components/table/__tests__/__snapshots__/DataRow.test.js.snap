// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataRow matches snapshot 1`] = `
<div
  class="css-g0yt5d-Row exk1gay0"
  data-row="2"
>
  <div
    class="row-header-width css-1i9qrac-Root e1d5mcq00"
    data-row="2"
  >
    2
    <button
      class="css-i54eap-BtnHeight e1d5mcq01"
      tabindex="-1"
    />
  </div>
  <div
    class="css-1kx7ize-Root-OutterBorder e17q3q8l0"
    data-cell="border"
  >
    <div
      class="css-4dsbpr-Root-Datum e1rvd88v0"
      data-cell="result"
      data-location="A-2"
      tabindex="0"
    />
  </div>
  <div
    class="css-1kx7ize-Root-OutterBorder e17q3q8l0"
    data-cell="border"
  >
    <div
      class="css-4dsbpr-Root-Datum e1rvd88v0"
      data-cell="result"
      data-location="B-2"
      tabindex="0"
    />
  </div>
</div>
`;
